globals/matchmaking_globals.go

```diff
@@ -9,4 +11,60 @@ type CommonMatchmakeSession struct {
 	ConnectionIDs []uint32 //players in the room, referenced by their connection IDs. This is used instead of the PID in order to ensure we're talking to the correct client (in case of e.g. multiple logins)
 }
 
-var Sessions = []CommonMatchmakeSession{}
\ No newline at end of file
+var Sessions = []CommonMatchmakeSession{}
+
+func DeleteIndex(s []uint32, index int) []uint32 {
+    return append(s[:index], s[index+1:]...)
+}
+
+func RemoveConnectionIDFromRoom(clientConnectionID uint32, gathering uint32){
+	for index, connectionID := range Sessions[gathering].ConnectionIDs {
+		if connectionID == clientConnectionID {
+			Sessions[gathering].ConnectionIDs = DeleteIndex(Sessions[gathering].ConnectionIDs, index)
+		}
+	}
+	if len(Sessions[gathering].ConnectionIDs) == 0{
+		Sessions[gathering] = CommonMatchmakeSession{} //TODO: is there a better way to do this? right now, the index is used as the gathering id, and I'm not sure if there's a good way to remove without reordering.
+	}
+}
+
+func FindClientSessions(clientConnectionID uint32) []uint32 {
+	//TODO: is there any real instance where a client could be in multiple MatchmakeSessions at once? afaik there isn't, but I handle it regardless just in case (if we're sure there isn't, we could remove the handling and speed the code up)
+	gatheringsFoundIn := []uint32{}
+	for gatheringIndex, _ := range Sessions {
+		for _, connectionID := range Sessions[gatheringIndex].ConnectionIDs {
+			if connectionID == clientConnectionID {
+				gatheringsFoundIn = append(gatheringsFoundIn, uint32(gatheringIndex))
+			}
+		}
+	}
+	return gatheringsFoundIn
+}
+
+func RemoveConnectionIDFromAllSessions(clientConnectionID uint32){
+	foundSessions := FindClientSessions(clientConnectionID)
+	for gatheringIndex, _ := range foundSessions {
```

Comment:
If you only need the index, a standard `for` loop is likely more performant since `for range` already reads the value at the index during iteration, which is thrown out here

Reply:
As in 
`gatheringIndex := 0; gatheringIndex <= len(foundSessions); gatheringIndex++` ? or just removing the _?

Reply:
```suggestion
	for i := 0; i < len(foundSessions); i++ {
```

(`i` can be changed that's just my usual default)